x-pack/plugins/security_solution/docs/testing/test_plans/detection_response/prebuilt_rules/installation_and_upgrade.md

```diff
@@ -815,6 +821,122 @@ And the Investigation Guide tab should NOT be displayed
 
 TODO: add scenarios https://github.com/elastic/kibana/issues/166215
 
+### Rule upgrade workflow: simple diff algorithm
+
+#### **Scenario: Rule field doesn't have an update and has no custom value**
+
+**Automation** 1 integration test
+
+```Gherkin
+Given at least 1 prebuilt rule is installed in Kibana
+And for this rule there is a new version available
+And the current version of <field> is unchanged from the base version
+And there is no update for <field> in this upgrade
+Then the merged output should be the current version of <field>
+And there should be no conflict
```

Comment:
I don't this it would be enough to check this.

This assertion checks only the result of the diff algorithm - something that we've already covered with unit tests (although for some reason we don't mention unit tests in the "Automation" comment -- why, by the way?).

I think what we need from a test plan on top of that is describing how the app should work at the API and UI levels. For instance, in this case, we could say:

```Gherkin
Then for <field_name> field the diff algorithm should output the current version as the merged one without a conflict
And <field_name> field should not be returned from the `upgrade/_review` API endpoint
And <field_name> field should not be shown in the upgrade preview UI
```

Then:

- The 1st "then" will be covered with unit tests (already done)
- The 2nd "then" will be covered with integration tests (in the next PR)
- The 3rd "then" will be covered with e2e tests (later, when we build the [interactive upgrade flyout](https://github.com/elastic/kibana/issues/171520))